Source/WebCore/platform/animation/AcceleratedEffectStack.h

```diff
@@ -59,22 +58,28 @@ class AcceleratedEffectStack {
     const AcceleratedEffectValues& baseValues() { return m_baseValues; }
     WEBCORE_EXPORT void setBaseValues(const AcceleratedEffectValues&&);
     
-    WEBCORE_EXPORT void applyPrimaryLayerEffects(PlatformLayer *, Seconds);
+    WEBCORE_EXPORT void initAsyncEffects(PlatformLayer *, Seconds);
     WEBCORE_EXPORT void applyBackdropLayerEffects(PlatformLayer *, Seconds) const;
+    WEBCORE_EXPORT void applyEffectsAsync(const FloatRect&, Seconds) const;
     WEBCORE_EXPORT void clear(PlatformLayer *);
```

Comment:
If we're going to have a base and remote version, we can have these methods only to the `RemoteAcceleratedEffectStack` since this is the only object that will actually deal with applying effects, correct? I think we can also simplify the class structure by keeping the base named as `AcceleratedEffectStack` and only subclass once for `RemoteAcceleratedEffectStack`.

Additionally, I think it would be best to keep the clear `primary` v. `backdrop` effects distinction.

Finally, since I don't think we will ever have a sync version, could we remove the `Async` part?

Reply:
The `Base` piece is to separate out the `WTF_MAKE_FAST_ALLOCATED` macro. I wanted to make the remote version refcounted, but that isn't possible if it inherits from a fast allocated class. An alternative would just to make `AcceleratedEffectStack` refcounted, and switch the existing callsites to use a `RefPtr`.

The `Async` naming was to imply that it was the one that gets called from another thread, whereas everything else is main-thread. It's not very effective.

Reply:
> The `Base` piece is to separate out the `WTF_MAKE_FAST_ALLOCATED` macro. I wanted to make the remote version refcounted, but that isn't possible if it inherits from a fast allocated class. An alternative would just to make `AcceleratedEffectStack` refcounted, and switch the existing callsites to use a `RefPtr`.

I see. We can make `AcceleratedEffectStack` ref-counted, that is fine. Unless there's a downside?

> The `Async` naming was to imply that it was the one that gets called from another thread, whereas everything else is main-thread. It's not very effective.

OK. I expect we must have a pattern for this, maybe name it `FromScrollingThread` or something.